api/src/domain/tax-filings/taxIdHelper.ts

```diff
@@ -0,0 +1,34 @@
+import { TaxFiling } from "@shared/taxFiling";
+import { TaxFilingResults } from "src/domain/types";
+
+export const reformatTaxId = (id: string) => id.trim().replaceAll(" ", "").replaceAll("/", "_").toLowerCase();
+
+export const getTaxIds = (id: string, map: Record<string, string[]> = {}): string[] => {
+  const val = reformatTaxId(id);
+  return map[val] ?? [val];
+};
+
+export const flattenDeDupAndMapTaxFilings = (
+  results: TaxFilingResults,
+  map?: Record<string, string[]>
+): TaxFiling[] =>
+  Object.values(
+    results?.reduce(
+      (accumulator: Record<string, TaxFiling[]>, result) => ({
+        ...accumulator,
+        ...getTaxIds(result.Id, map).reduce(
```

Comment:
a reduce within a reduce? 😳 

could this be rewritten with more helper functions and intermediate variables that have descriptive names, so that someone else would be able to look at this and understand how it works? There's no reason for this to be a one-liner, and it's truly impossible to parse. 

A good heuristic is to write something imagining the most junior member of our team coming in and trying to read it / make a change, and writing in a way that will enable them to do that without help or without talking to you to explain it.